lib/createGltf.js

```diff
@@ -372,40 +356,41 @@ function createMetallicRoughnessTexture(gltf, metallicImage, roughnessImage, occ
     var height = dimensions[1];
     var pixelsLength = width * height;
     var pixels = Buffer.alloc(pixelsLength * 4, 0xFF); // Initialize with 4 channels, unused channels will be white
+    var scratchChannel = Buffer.alloc(pixelsLength);
 
     if (packMetallic) {
         // Write into the B channel
-        var metallicChannel = getImageChannel(metallicImage, 0, width, height);
-        writeChannel(pixels, metallicChannel, 2, width, height);
+        var metallicChannel = getImageChannel(metallicImage, 0, width, height, scratchChannel);
```

Comment:
Since it's possible that the returned channel is a newly allocated buffer, should this (and below) update `scratchChannel` as well? Otherwise this might still cause repeated reallocation in some cases. Seems like it's "fine" if `scratchChannel` is bigger than it needs to be.